custom/icds/location_reassignment/parser.py

```diff
@@ -151,3 +153,36 @@ def _validate_descendants_archived(self):
                 self.errors.append("Location %s is getting archived but the following descendants are not %s" % (
                     location.site_code, ",".join(missing_site_codes)
                 ))
+
+
+class HouseholdReassignmentParser(object):
+    def __init__(self, domain, workbook):
+        self.domain = domain
+        self.workbook = workbook
+        self.reassignments = {}  # household id mapped to new awc code
+
+    def parse(self):
+        errors = []
+        site_codes = set()
+        for worksheet in self.workbook.worksheets:
+            location_site_code = worksheet.title
+            site_codes.add(location_site_code)
+            for row in worksheet:
+                household_id = row.get(HOUSEHOLD_ID_COLUMN)
+                new_awc_code = row.get(AWC_CODE_COLUMN)
+                if not household_id:
+                    errors.append("Missing Household ID")
+                if not household_id and not new_awc_code:
+                    errors.append("Missing New AWC Code")
+                site_codes.add(new_awc_code)
+                self.reassignments[household_id] = {
+                    'old_site_code': location_site_code,
+                    'new_site_code': new_awc_code
+                }
+        locations = SQLLocation.active_objects.filter(site_codes)
```

Comment:
Doesn't `filter` need a named parameter, like `site_code__in=[site_codes]`?

Reply:
https://github.com/dimagi/commcare-hq/pull/27134/commits/a9370dde45443e42086ee3e4cff5feaba2618bf0

Reply:
https://github.com/dimagi/commcare-hq/pull/27134/commits/5c0594602dca8c4257de6c57e23281bc275390f8

Reply:
thanks for catching that.